packages/dds/tree/src/feature-libraries/chunked-forest/codec/nodeShape.ts

```diff
@@ -39,6 +40,24 @@ export class NodeShape extends Shape<EncodedChunkShape> implements NodeEncoder {
 		this.explicitKeys = new Set(this.fields.map((f) => f.key));
 	}
 
+	private getEncodeValue(cursor: ITreeCursorSynchronous, cache: EncoderCache): Value {
+		if (this.value === 0) {
+			assert(typeof cursor.value === "string", "identifier must be type string");
+			if (
+				isStableId(cursor.value) &&
+				cache.idCompressor.tryRecompress(cursor.value) !== undefined
+			) {
+				const sessionSpaceCompressedId = cache.idCompressor.tryRecompress(cursor.value);
+				assert(
```

Comment:
Is this correct? We weren't asserting before. What happens if you pass in a string that was not generated by the compressor, but is a UUID? Like `b3767cd8-3dfa-451b-83d8-36571ad6b72b`? In that case, we want to fallback to the default case (return cursor.value), not fail - right? If that's true, there should be another unit test added for that case as well.

Reply:
Oh - nevermind, I see you are calling `tryRecompress` twice. Instead, just make it another `if` conditional and only do it once.